### PR TITLE
Make shadow traffic request and send sync\rTESTING=unit

### PR DESCRIPTION
**Pull request by scottmcmaster**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -5,7 +5,8 @@ git_source(:github) {|repo_name| "https://github.com/#{repo_name}" }
 # Specify your gem's dependencies in promoted-ruby-client.gemspec
 gemspec
 
-gem 'faraday', '~> 1.4.1'
+gem 'async-http'
```
Comment (danbosnichill) [inline Gemfile:8]: I sent some direct chats about this.  Yea, I think eventually having a plugin for this makes sense.

File: lib/promoted/ruby/client.rb [hunk 2]
```diff
@@ -1,13 +1,12 @@
...
-require 'faraday'
-require 'json'
 
 module Promoted
   module Ruby
     module Client
 
       DEFAULT_DELIVERY_TIMEOUT_MILLIS = 3000
-      DEFAULT_METRICS_TIMEOUT_MILLIS = 250
+      DEFAULT_METRICS_TIMEOUT_MILLIS = 3000
```
Comment (danbosnichill) [inline lib/promoted/ruby/client.rb:9]: Ah, true.  The timeouts are flipped.  We should have shorter timeouts for Delivery API.

**Review by danbosnichill — APPROVED** [review]
We're still talking about this in Slack.

File: lib/promoted/ruby/client.rb (hunk 2 above)
Comment (scottmcmaster) [reply lib/promoted/ruby/client.rb:9]: The unit of timeout we have to work with here is in seconds, not millis, though.

Comment (scottmcmaster) [reply lib/promoted/ruby/client.rb:9]: So bare minimum is 1000 millis, or else the request never even gets sent in my testing.

**Pull request merged**